pkg/apb/svc_acct.go

```diff
@@ -0,0 +1,197 @@
+				"name":      svcAccountName,
+				"namespace": namespace,
+			},
+		},
+		"roleRef": map[string]string{
+			"name": apbRole,
+		},
+	}
+
+	s.createResourceDir()
+	rFilePath, err := s.writeResourceFile(apbId, &svcAcctM, &roleBindingM)
+	if err != nil {
+		return "", err
+	}
+
+	// Create resources in cluster
+	s.createResources(rFilePath, namespace)
+
+	s.log.Info("Successfully created apb sandbox: [ %s ]", apbId)
+
+	return apbId, nil
+}
+
+func (s *ServiceAccountManager) createResources(rFilePath string, namespace string) error {
+	s.log.Debug("Creating resources from file at path: %s", rFilePath)
+	output, err := RunCommand("oc", "create", "-f", rFilePath, "--namespace="+namespace)
+	// TODO: Parse output somehow to validate things got created?
+	if err != nil {
+		s.log.Error("Something went wrong trying to create resources in cluster")
+		s.log.Error("Returned error:")
+		s.log.Error(err.Error())
+		s.log.Error("oc create -f output:")
+		s.log.Error(string(output))
+		return err
+	}
+	s.log.Debug("Successfully created resources, oc create -f output:")
+	s.log.Debug("\n" + string(output))
+	return nil
+}
+
+func (s *ServiceAccountManager) writeResourceFile(
+	apbId string,
+	svcAcctM *map[string]interface{},
+	roleBindingM *map[string]interface{},
+) (string, error) {
+	// Create file if doesn't already exist
+	filePath := s.createFile(apbId)
+
+	file, err := os.OpenFile(filePath, os.O_RDWR, 0644)
+	defer file.Close()
+
+	if err != nil {
+		s.log.Error("Something went wrong writing resources to file!")
+		s.log.Error(err.Error())
+		return "", err
+	}
+
+	file.WriteString("---\n")
+	svcAcctY, err := yaml.Marshal(svcAcctM)
+	if err != nil {
+		s.log.Error("Something went wrong marshalling svc acct to yaml")
+		s.log.Error(err.Error())
+		return "", err
+	}
+	file.WriteString(string(svcAcctY))
+
+	file.WriteString("---\n")
+	roleBindingY, err := yaml.Marshal(roleBindingM)
+	if err != nil {
+		s.log.Error("Something went wrong marshalling role binding to yaml")
+		s.log.Error(err.Error())
+		return "", err
+	}
+	file.WriteString(string(roleBindingY))
+
+	s.log.Info("Successfully wrote resources to %s", filePath)
+	return filePath, nil
+}
+
+func (s *ServiceAccountManager) createResourceDir() {
+	dir := resourceDir()
+	s.log.Debug("Creating resource file dir: %s", dir)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		os.Mkdir(dir, os.ModePerm)
+	}
+}
+
+func (s *ServiceAccountManager) createFile(apbId string) string {
+	rFilePath := filepath.Join(resourceDir(), apbId+".yaml")
+	s.log.Debug("Creating resource file %s", rFilePath)
+	var _, err = os.Stat(rFilePath)
+
+	if os.IsNotExist(err) {
+		var file, err = os.Create(rFilePath)
+		if err != nil {
+			s.log.Error("Something went wrong touching resource file!")
+			s.log.Error(err.Error())
+		}
+		defer file.Close()
+	}
+	return rFilePath
+}
+
+func (s *ServiceAccountManager) DestroyApbSandbox(
+	handle string,
+	namespace string,
+) error {
+	if handle == "" {
```

Comment:
Yes, any of the apb methods that run but have error'd out before creating the sandbox return "" for the podName. I'm not happy with it, but this is something I want to come back to after TP1 to do sandbox destruction the right way (some kind of pod termination event handler).